File: src/services/library.service.ts
import { Book } from '../models/book.model'
import { Customer } from '../models/customer.model'

export interface LibraryService {
  readonly maxLendingPeriodInWeeks: number

  getBooks(): Book[]

  getAvailableBooks(): Book[]

  getBooksByTitle(title: string): Book[]

  getAvailableBooksByTitle(title: string): Book[]

  getLentBooksByCustomer(customer: Customer): Book[]

  getOverdueBooks(): Book[]
}

export const libraryService = {} as LibraryService

File: src/main/kotlin/services/LibraryService.kt
package services

import models.Book
import models.Customer

interface LibraryService {
    val maxLendingPeriodInWeeks: Int

    fun getBooks(): List<Book>

    fun getAvailableBooks(): List<Book>

    fun getBooksByTitle(title: String): List<Book>

    fun getAvailableBooksByTitle(title: String): List<Book>

    fun getLentBooksByCustomer(customer: Customer): List<Book>

    fun getOverdueBooks(): List<Book>
}
